mmgen/models/architectures/biggan/generator_discriminator_deep.py

```diff
@@ -0,0 +1,635 @@
+                Defaults to None.
+            init_type (str, optional): The name of an initialization method:
+                ortho | N02 | xavier. Defaults to 'ortho'.
+        """
+        if isinstance(pretrained, str):
+            logger = get_root_logger()
+            load_checkpoint(self, pretrained, strict=False, logger=logger)
+        elif isinstance(pretrained, dict):
+            ckpt_path = pretrained.get('ckpt_path', None)
+            assert ckpt_path is not None
+            prefix = pretrained.get('prefix', '')
+            map_location = pretrained.get('map_location', 'cpu')
+            strict = pretrained.get('strict', True)
+            state_dict = _load_checkpoint_with_prefix(prefix, ckpt_path,
+                                                      map_location)
+            self.load_state_dict(state_dict, strict=strict)
+            mmcv.print_log(f'Load pretrained model from {ckpt_path}', 'mmgen')
+        elif pretrained is None:
+            for m in self.modules():
+                if isinstance(m, (nn.Conv2d, nn.Linear, nn.Embedding)):
+                    if init_type == 'ortho':
+                        nn.init.orthogonal_(m.weight)
+                    elif init_type == 'N02':
+                        normal_init(m, 0.0, 0.02)
+                    elif init_type == 'xavier':
+                        xavier_init(m)
+                    else:
+                        raise NotImplementedError(
+                            f'{init_type} initialization \
+                            not supported now.')
+        else:
+            raise TypeError('pretrained must be a str or None but'
+                            f' got {type(pretrained)} instead.')
+
+
+@MODULES.register_module()
+class BigGANDeepDiscriminator(nn.Module):
+    """BigGAN-Deep Discriminator.
+
+    Args:
+        input_scale (int): The scale of the input image.
+        num_classes (int, optional): The number of conditional classes.
+            Defaults to 0.
+        in_channels (int, optional): The channel number of the input image.
+            Defaults to 3.
+        out_channels (int, optional): The channel number of the final output.
+            Defaults to 1.
+        base_channels (int, optional): The basic channel number of the
+            discriminator. The other layers contains channels based on this
+            number. Defaults to 96.
+        block_depth (int, optional): The repeat times of Residual Blocks in
+            each level of architecture. Defaults to 2.
+        sn_eps (float, optional): Epsilon value for spectral normalization.
+            Defaults to 1e-6.
+        init_type (str, optional): The name of an initialization method:
+            ortho | N02 | xavier. Defaults to 'ortho'.
+        act_cfg (dict, optional): Config for the activation layer.
+            Defaults to dict(type='ReLU').
+        conv_cfg (dict, optional): Config for the convolution module used in
```

Comment:
This argument is a little confusing and may be removed if we have never changed it.

Reply:
Which argument? If you mean `conv_cfg`, I will remove it soon.

Reply:
Yep, it's `conv_cfg`.